src/grt/src/MakeWireParasitics.cpp

```diff
@@ -114,54 +128,155 @@ sta::Pin* MakeWireParasitics::staPin(Pin& pin)
+void MakeWireParasitics::makeWireParasitics(odb::dbNet* net,
+                        std::vector<GSegment>& routes,
+                        int& routes_cnt)
 {
+  sta::Units* units = sta_->units();
   for (GSegment& route : routes) {
-    sta::ParasiticNode* n1
-        = ensureParasiticNode(route.init_x, route.init_y, route.init_layer);
-    sta::ParasiticNode* n2
-        = ensureParasiticNode(route.final_x, route.final_y, route.final_layer);
     int wire_length_dbu = route.length();
-    sta::Units* units = sta_->units();
-    float res = 0.0;
-    float cap = 0.0;
-    if (wire_length_dbu == 0) {
-      // via
-      int lower_layer = min(route.init_layer, route.final_layer);
-      odb::dbTechLayer* cut_layer
-          = tech_->findRoutingLayer(lower_layer)->getUpperLayer();
-      res = cut_layer->getResistance();  // assumes single cut
-      cap = 0.0;
-      debugPrint(logger_,
-                 GRT,
-                 "est_rc",
-                 1,
-                 "{} -> {} via r={}",
-                 parasitics_->name(n1),
-                 parasitics_->name(n2),
-                 units->resistanceUnit()->asString(res));
-    } else if (route.init_layer == route.final_layer) {
+    if (wire_length_dbu > 0 &&
+        route.init_layer == route.final_layer) {
+      sta::ParasiticNode* n1
+        = ensureParasiticNode(route.init_x, route.init_y, route.init_layer);
+      sta::ParasiticNode* n2
+          = ensureParasiticNode(route.final_x, route.final_y, route.final_layer);
+
+      RoutePt n1_pt(route.init_x, route.init_y, route.init_layer);
+      RoutePt n2_pt(route.final_x, route.final_y, route.final_layer);
+
+      float res = 0.0;
+      float cap = 0.0;
       layerRC(wire_length_dbu, route.init_layer, res, cap);
+
+      // create the wire resistor
+      parasitics_->incrCap(n1, cap / 2.0, analysis_point_);
+      parasitics_->makeResistor(nullptr, n1, n2, res, analysis_point_);
+      parasitics_->incrCap(n2, cap / 2.0, analysis_point_);
       debugPrint(logger_,
                  GRT,
                  "est_rc",
                  1,
-                 "{} -> {} {}u layer={} r={} c={}",
+                 "(wire) {} -> {} {}u layer={} r={} c={}",
                  parasitics_->name(n1),
                  parasitics_->name(n2),
                  dbuToMeters(wire_length_dbu) * 1e+6,
                  route.init_layer,
                  units->resistanceUnit()->asString(res),
                  units->capacitanceUnit()->asString(cap));
-    } else
-      logger_->warn(GRT,
-                    25,
-                    "Non wire or via route found on net {}.",
-                    net->getConstName());
-    parasitics_->incrCap(n1, cap / 2.0, analysis_point_);
-    parasitics_->makeResistor(nullptr, n1, n2, res, analysis_point_);
-    parasitics_->incrCap(n2, cap / 2.0, analysis_point_);
+
+      // create resistors for pins connected directly to wires
```

Comment:
What happens for planar access to pins that don't go through vias?  I'm confused by this code.

Reply:
When we call `estimate_parasitcs -global_routing`, the pins are already covered by a guide to having the planar access. So it is not connected through vias, but it is connected directly to the wire, and this piece of code covers this case.

This code looks at the endpoints of each wire for pin nodes that share the same location (x, y, l), and connects these pins directly to the corresponding endpoint.